packages/base/src/commands.ts

```diff
@@ -285,6 +290,190 @@ export function addCommands(
     ...icons.get(CommandIDs.newVectorTileEntry)
   });
 
+  commands.addCommand(CommandIDs.buffer, {
+    label: trans.__('Buffer'),
+    isEnabled: () => {
+      const model = tracker.currentWidget?.model;
+      const localState = model?.sharedModel.awareness.getLocalState();
+
+      if (!model || !localState || !localState['selected']?.value) {
+        return false;
+      }
+
+      const selectedLayers = localState['selected'].value;
+
+      // Can't open more than one symbology dialog at once
+      if (Object.keys(selectedLayers).length > 1) {
+        return false;
+      }
+
+      const layerId = Object.keys(selectedLayers)[0];
+      const layer = model.getLayer(layerId);
+
+      if (!layer) {
+        return false;
+      }
+
+      const isValidLayer = ['VectorLayer', 'ShapefileLayer'].includes(
+        layer.type
+      );
+
+      return isValidLayer;
+    },
+    execute: async () => {
+      const layers = tracker.currentWidget?.model.sharedModel.layers ?? {};
+      const sources = tracker.currentWidget?.model.sharedModel.sources ?? {};
+
+      const layerOptions = Object.keys(layers).map(layerId => ({
+        value: layerId,
+        label: layers[layerId].name
+      }));
+
+      if (layerOptions.length === 0) {
+        console.warn('No layers available to buffer.');
+        return;
+      }
+
+      const schema = {
+        ...(formSchemaRegistry.getSchemas().get('Buffer') as IDict)
+      };
+
+      const model = tracker.currentWidget?.model;
+      const localState = model?.sharedModel.awareness.getLocalState();
+      if (!model || !localState || !localState['selected']?.value) {
+        return;
+      }
+      const localSelectedLayer = localState['selected'].value;
+      const localSelectedLayerId = Object.keys(localSelectedLayer)[0];
+
+      // Open form and get user input
+      const formValues = await new Promise<IDict>(resolve => {
+        const dialog = new ProcessingFormDialog({
+          title: 'Buffer',
+          schema: schema,
+          model: tracker.currentWidget?.model as IJupyterGISModel,
+          sourceData: {
+            inputLayer: localSelectedLayerId,
+            bufferDistance: 10,
+            projection: 'EPSG:4326'
+          },
+          cancelButton: false,
+          syncData: (props: IDict) => {
+            resolve(props);
+            dialog.dispose();
+          }
+        });
+
+        dialog.launch();
+      });
+
+      if (!formValues) {
+        return;
+      }
+
+      const bufferDistance = formValues.bufferDistance;
+      const selectedLayerId = formValues.inputLayer;
+      const selectedLayer = layers[selectedLayerId];
+
+      if (!selectedLayer.parameters) {
+        console.error('Selected layer not found.');
+        return;
+      }
+
```

Comment:
Maybe do this check before showing the dialog, since there is no point showing the dialog if we can't do anything after that.

Reply:
Ah right! Maybe rename `selectedLayer` into `inputLayer` then, to make things clearer (it may not be anymore the selected layer at this point)

Reply:
Sure, done in [12ac8a0](https://github.com/geojupyter/jupytergis/pull/510/commits/12ac8a012367863155c9f68831d16dc8011f7181)